### PR TITLE
Add columns to drivers

### PR DESCRIPTION
**Pull request by bhkhdh**
It's for issue  #23

### CODE REVIEW

File: app/models/admin.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class Admin < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
-  devise :database_authenticatable,
+  devise :database_authenticatable, :registerable,
```
Comment (rakvium) [inline app/models/admin.rb]: Why registerable?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -8,16 +8,27 @@
...
+<%= link_to("Register new admin", new_admin_registration_path) %>
+<%= link_to("Register new dispatcher", new_dispatcher_registration_path) %>
 <%= link_to("Register new driver", new_driver_registration_path) %>
-<%elsif !current_driver %>
+<%elsif !current_driver && !current_dispatcher %>
 <%= link_to("Admins: Log in", admin_session_path) %>
 <%end%>
+
+<% if  current_dispatcher %>
+<%= link_to("Log out", destroy_dispatcher_session_path) %>
```
Comment (rakvium) [inline app/views/layouts/application.html.erb]: What if we have both driver and dispatcher logged in?

File: config/routes.rb [hunk 3]
```diff
@@ -1,25 +1,10 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  devise_for :admins
+  devise_for :drivers,  :controllers => { registrations: 'registrations' }
```
Comment (rakvium) [inline config/routes.rb]: Please use new Ruby hash literal syntax: https://github.com/bbatsov/ruby-style-guide#hash-literals

**Review by rakvium — CHANGES_REQUESTED** [review]

File: app/models/admin.rb (hunk 1 above)
Comment (bhkhdh) [reply app/models/admin.rb]: By the fact that the administrator can register another administrator, and if remove "regitrable", they can't do this.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (bhkhdh) [reply app/views/layouts/application.html.erb]: I I'm working on it

File: config/routes.rb (hunk 3 above)
Comment (bhkhdh) [reply config/routes.rb]: It's doesn't working when i write 'registrations => 'registrations''. Main trget of this request is adding colomn :D

File: app/models/admin.rb (hunk 1 above)
Comment (rakvium) [reply app/models/admin.rb]: "Registerable" there means that any user can come and register himself as an admin. It's not an option.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (rakvium) [reply app/views/layouts/application.html.erb]: OK, just add `as a dispatcher` to eliminate an ambiguity :relieved:

File: config/routes.rb (hunk 3 above)
Comment (rakvium) [reply config/routes.rb]: There should be `controllers:`, not `:controllers =>`.

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -39,14 +39,13 @@ class Driver::RegistrationsController < Devise::RegistrationsController
...
-  end
+   def sign_up_params
+    params.require(:driver).permit(:email, :password, :password_confirmation, :name, :phone_number_string, :license_plate, :car_model, :car_color, :car_type, :upload => [:tempfile, :original_filename, :content_type, :headers])
    end
 
   # If you have extra params to permit, append them to the sanitizer.
-   def configure_account_update_params
-    params.require(:user).permit(:email, :password, :password_confirmation, :license_plate, :car_model, :car_color, :car_type, :current_pussword)
+   def update_params
+    params.require(:driver).permit(:email, :password, :password_confirmation, :current_pussword, :name, :phone_number_string, :license_plate, :car_model, :car_color, :car_type, :upload => [:tempfile, :original_filename, :content_type, :headers])
```
Comment (rakvium) [inline app/controllers/registrations_controller.rb]: `current_pussword`? `pussword`? Seriously?

**Review by rakvium — CHANGES_REQUESTED** [review]

File: app/models/ability.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
```
Comment (rakvium) [inline app/models/ability.rb:5]: You have no User model in the project :confused:

**Review by rakvium — CHANGES_REQUESTED** [review]

File: app/models/ability.rb (hunk 5 above)
Comment (bhkhdh) [reply app/models/ability.rb:5]: It's fixed if branch "cancancan"

**Review by rakvium — APPROVED** [review]

**Pull request merged**